### PR TITLE
Revert 3744

### PR DESCRIPTION
**Pull request by jbartosik**
* Revert #3744 because I no longer think we need to make the change.
* Add a comment explaining why we filter pods the way we do so if someone in the future wants to change the behaviour they need to do less archeologi

### CODE REVIEW

File: vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go [hunk 1]
```diff
@@ -175,11 +175,11 @@ func newPodClients(kubeClient kube_client.Interface, resourceEventHandler cache.
...
 	// running but there are some transient errors we don't want to delete it from
 	// our model).
-	// We don't want to watch Pending, Succeeded, or Failed failed pods because we
-	// know they're not generating any usage.
-	// Filter
-	statusFilter := fmt.Sprintf("status.phase!=%s,status.phase!=%s,status.phase!=%s", apiv1.PodPending, apiv1.PodSucceeded, apiv1.PodFailed)
-	selector := fields.ParseSelectorOrDie(statusFilter)
+	// We don't want to watch Pending pods because they didn't generate any usage
+	// yet.
+	// Succeeded and Failed failed pods don't generate any usage anymore but they
```
Comment (bskiba) [inline vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go:180]: Can you rephrase the comment to say what data we lose when we stopped listing those pods? 
It's also a bit worrying since I'm not sure if API server gives any guarantees on how long the failed and succeeded pods are kept for.

Comment (jbartosik) [reply vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go]: Rephrased. I didn't see any deleted data (that we wouldn't want to delete). I think we won't delete any. I just want to highlight here that one needs to be careful if they decide to change this

**Comment by jbartosik** [comment]
@bskiba please take a look. I updated cherry pick  (#3809) too.
